### PR TITLE
Document breadcrumbs interface

### PR DESCRIPTION
**Pull request by benvinegar**
Note I intentionally left off the "query" and "rpc" breadcrumbs because we haven't represented them in the UI yet.

I also left off the classifier stuff because I don't think we've fleshed that out either.

cc @mitsuhiko @dcramer @nateberkopec

### CODE REVIEW

**Comment by benvinegar** [comment]
Example of how the individual breadcrumb type descriptions are rendered inside docs:

image

File: docs/clientdev/interfaces.rst [hunk 1]
```diff
@@ -329,3 +329,175 @@ data related to the current user, the current HTTP request.
...
+        }, {
+          "timestamp": 1461185753847,
+          "type": "navigation",
+          "data": {
+            "from": "/login",
+            "to": "/dashboard"
+          }
+        }]
+
+Below are descriptions of individual breadcrumb types, and what their ``data`` properties look like.
```
Comment (dcramer) [inline docs/clientdev/interfaces.rst]: I would add a Types header

File: docs/clientdev/interfaces.rst [hunk 2]
```diff
@@ -329,3 +329,175 @@ data related to the current user, the current HTTP request.
...
+
+.. describe:: "message" breadcrumb
```
Comment (dcramer) [inline docs/clientdev/interfaces.rst]: just call it "message" here instead of saying its a '"message" breadcrumb'

**Comment by benvinegar** [comment]
Updated.

@dcramer – did you want me to push this and deploy docs, or is it sufficient to just point people at the PR for now?

**Comment by dcramer** [comment]
lgtm

**Pull request merged**